src/dscanner/main.d

```diff
@@ -270,29 +270,39 @@ else
 	{
 		if (sloc || tokenCount)
 		{
+			import dmd.lexer : Lexer;
+			import dmd.tokens : TOK;
+			import std.string : toStringz;
+			import core.stdc.string : strlen;
+
 			if (usingStdin)
 			{
-				LexerConfig config;
-				config.stringBehavior = StringBehavior.source;
-				auto tokens = byToken(readStdin(), config, &cache);
+				auto sourceDstr = readStdin();
+				auto sourceCstr = toStringz(cast(char[]) sourceDstr);
+
+				Lexer lexer = new Lexer(null, sourceCstr, 0, strlen(sourceCstr), false, true, true);
```

Comment:
There is no need to compute the length of the string here. `sourceDstr` already contains that information (the length property).

Reply:
Also, I don't like all those magic variables `false, true, true` ... what do they mean?